### PR TITLE
dotCMS/core#16734 Internationalization (i18n) dot-form and fields

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/cache/KeyValueCache.java [hunk 1]
```diff
@@ -57,6 +57,19 @@
      */
     public void addByLanguageAndContentType(final long languageId, final String contentTypeId, final KeyValue keyValue);
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/cache/KeyValueCache.java]: needs test

File: dotCMS/src/main/java/com/dotcms/cache/KeyValueCache.java [hunk 2]
```diff
@@ -97,6 +110,24 @@
      */
     public KeyValue getByLanguageAndContentType(final String key, final long languageId, final String contentTypeId);
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/cache/KeyValueCache.java]: needs test

File: dotCMS/src/main/java/com/dotcms/contenttype/model/field/layout/FieldLayoutColumnSerializer.java [hunk 3]
```diff
@@ -39,6 +43,9 @@ public void serialize(
             final JsonGenerator jsonGenerator,
             final SerializerProvider serializerProvider) throws IOException {
 
+        final ContentTypeInternationalization contentTypeInternationalization =
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/field/layout/FieldLayoutColumnSerializer.java]: needs test

File: dotCMS/src/main/java/com/dotcms/contenttype/model/field/layout/FieldUtil.java [hunk 4]
```diff
@@ -231,6 +236,27 @@ public static Field copyField(final Field field, final int newSortOrder) {
         return jsonFieldTransformer.from();
     }
 
+    public static void setFieldInternationalization(
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/field/layout/FieldUtil.java]: needs test

File: dotCMS/src/main/java/com/dotcms/keyvalue/business/KeyValueAPIImpl.java [hunk 5]
```diff
@@ -120,38 +120,42 @@ public KeyValueAPIImpl() {
     public KeyValue get(final String key, final long languageId, final ContentType contentType, final User user,
                     final boolean respectFrontendRoles) {
 
+        return this.get(key, languageId, contentType, user, true, respectFrontendRoles);
+    }
+
+    @Override
+    public KeyValue get(final String key, final long languageId, final ContentType contentType, final User user, final boolean live,
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/keyvalue/business/KeyValueAPIImpl.java]: needs test

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/StructuresWebAPI.java [hunk 6]
```diff
@@ -273,11 +277,24 @@ public boolean isFieldHidden(final Field field) throws DotDataException{
 	 */
 	public String getLayoutAsJson(final Structure structure, final User user) {
 		try {
+			String languageId = request.getParameter("languageId");
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/StructuresWebAPI.java]: Always use the `LanguageWebAPI` to resolve front end language

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeResource.java [hunk 7]
```diff
@@ -262,7 +260,12 @@ public Response deleteType(@PathParam("idOrVar") final String idOrVar, @Context
 	@JSONP
 	@NoCache
 	@Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-	public Response getType(@PathParam("idOrVar") final String idOrVar, @Context final HttpServletRequest req, @Context final HttpServletResponse res)
+	public Response getType(
+			@PathParam("idOrVar") final String idOrVar,
+			@Context final HttpServletRequest req,
+			@Context final HttpServletResponse res,
+			@QueryParam("language_id") final Long languageId,
```
Comment (jgambarios) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/ContentTypeResource.java]: In order to follow the same naming pattern of the method we could change the parameter to `languageId`

**Review by jgambarios — CHANGES_REQUESTED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**